apps/wallet-ui/src/app/[ecosystem]/(ui)/wallet/[address]/layout.tsx

```diff
@@ -0,0 +1,58 @@
+import { ChainCombobox } from "@/components/ChainCombobox";
+import { client } from "@/lib/client";
+import { getEcosystemInfo } from "@/lib/ecosystems";
+import type { Metadata, ResolvingMetadata } from "next";
+import { resolveName } from "thirdweb/extensions/ens";
+import { shortenAddress } from "thirdweb/utils";
+import { getChains } from "../../../../../lib/chains";
+import { SIMPLEHASH_NFT_SUPPORTED_CHAIN_IDS } from "../../../../../util/simplehash";
+
+export async function generateMetadata(
+  { params }: { params: { ecosystem: string; address: string } },
+  parent: ResolvingMetadata,
+): Promise<Metadata> {
+  const ecosystem = await getEcosystemInfo(params.ecosystem);
+
+  const previousImages = (await parent).openGraph?.images || [];
```

Comment:
can call these in parallel, also does it not merge the OG images for you for nested routes - feels like it should?

Reply:
Took this directly from the next.js docs, you have to explicitly merge if openGraph.images is defined afaik